crates/red_knot_python_semantic/src/types/type_ordering.rs

```diff
@@ -118,6 +118,9 @@ pub(super) fn union_or_intersection_elements_ordering<'db>(
         (Type::Instance(_), _) => Ordering::Less,
         (_, Type::Instance(_)) => Ordering::Greater,
 
+        (Type::TypeVar(_), _) => Ordering::Less,
```

Comment:
I think you need to also make sure that two different TypeVars always have a consistent ordering here:

```suggestion
        (Type::TypeVar(left), Type::TypeVar(right)) => left.cmp(right),
        (Type::TypeVar(_), _) => Ordering::Less,
```